test/helpers/position_assertions.h

```diff
@@ -89,7 +90,21 @@ class ErrorAssertion final : public RangeAssertion {
 
     std::string toString() const override;
 
-    bool check(const Diagnostic &diagnostic, std::string_view sourceLine, std::string_view errorPrefix);
+    virtual bool check(const Diagnostic &diagnostic, std::string_view sourceLine, std::string_view errorPrefix);
+};
+
+class InfoAssertion final : public ErrorAssertion {
+public:
+    static std::shared_ptr<InfoAssertion> make(std::string_view filename, std::unique_ptr<Range> &range,
+                                               int assertionLine, std::string_view assertionContents,
+                                               std::string_view assertionType);
+
+    InfoAssertion(std::string_view filename, std::unique_ptr<Range> &range, int assertionLine,
+                  std::string_view message);
+
+    std::string toString() const override;
+
+    bool check(const Diagnostic &diagnostic, std::string_view sourceLine, std::string_view errorPrefix) override;
```

Comment:
An alternative implementation would be to just have ErrorAssertion and have an enum to specify the assertion type (as opposed to current boolean flag for error-with-dupes vs without) and then do casework within it.

This, however, follows the philosophy of there are only errors, and this highlighting information diagnostic is a separate thing unto itself. Adds a virtual function call though. Doing it without deriving from ErrorAssertion would introduce a huge amount of code duplication.

Reply:
On second thought, the tests are failing when the harness encounters this assertion in the non-LSP powered tests (we have normal pipeline tests, that don't start up LSP).

So in that case, it might actually be useful to have a separate assertion like "# ^^^ untyped" which can be ignored in the non-LSP tests.